app/components/ny_times/builder.rb

```diff
@@ -0,0 +1,67 @@
+module NyTimes
+  # A builder for articles that come from the New York Times
+  class Builder
+    attr_reader :article
+    def initialize(response_body, article_path, article_id = nil)
+      @article_path = article_path
+      @response_body = response_body
+      parse_response
+      article_hash = build_article_hash
+      @article =
+        if article_id
+          update_article(article_hash, article_id)
+        else
+          create_article(article_hash)
+        end
+    end
+
+    # Builds a Hase with `Article` params
+    #
+    # @return [Hash] article params
+    def build_article_hash
+      {
+        title: article_title,
+        publication: new_york_times,
+        date: article_published_on,
+        excerpt: article_excerpt(article_body),
+        link: "https://www.nytimes.com#{@article_path}"
+      }
+    end
+
+    private
+
+    def update_article(article_hash, article_id)
```

Comment:
Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)